src/Builder/Type/OperatorInterface.php

```diff
@@ -9,5 +9,8 @@
  */
 interface OperatorInterface
 {
+    /** To be overridden by implementing classes */
+    public const ENCODE = Encode::Single;
```

Comment:
Noted that you're defaulting to `Single` here instead of introducing an invalid case. My thinking was the invalid case would make it easier to catch mistakes when failing to override this, since `Single` would just happen to work (at least for encoding) provided the class only had a single property.

Reply:
You can add a `Undefined` case to the enum.

Reply:
I'm personally a fan of "Undefined" but won't push for it.

Reply:
Introduced `Undefined` value to the enum. Agree with not abusing `null` for this to keep the constant strictly typed.